tests/expressions/tests.py

```diff
@@ -534,6 +534,33 @@ def test_subquery_references_joined_table_twice(self):
         outer = Company.objects.filter(pk__in=Subquery(inner.values('pk')))
         self.assertFalse(outer.exists())
 
+    def test_filter_by_count_annotated_inside_subquery(self):
```

Comment:
Let's make the test name aggregate function agnostic, what about the following?

```suggestion
    def test_subquery_filter_by_aggregate(self):
```